spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -0,0 +1,418 @@
+      end
+
+    end
+
+    context 'when the response is nil' do
+
+      it 'returns false' do
+        expect(subject.abs_is_valid_response?(nil)).to eq(false)
+      end
+
+    end
+
+    context 'when the response code is not valid' do
+
+      it 'returns false' do
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_INVALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_AWSDIRECT_RESPONSE_BODY)
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+    context 'when the response body is empty' do
+
+      it 'returns false' do
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_INVALID_RESPONSE_BODY)
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+    context 'when the response body is nil' do
+
+      it 'returns false' do
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(nil)
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+
+  end
+
+  describe '#abs_reformat_resource_host' do
+
+    context 'when a valid response_body is provided' do
+
+      it 'returns the expected reformatted response' do
+        expect(subject.abs_reformat_resource_host(TEST_AWSDIRECT_RESPONSE_BODY)).to eq(TEST_REFORMATTED_RESPONSE_BODY)
+      end
+
+    end
+
+    context 'when an invalid response_body is provided' do
+
+      it 'reports the error and returns nil' do
+        expect(subject).to receive(:puts).with('JSON::ParserError encountered')
+        expect(subject.abs_reformat_resource_host(TEST_INVALID_RESPONSE_BODY)).to eq(nil)
+
+      end
+
+    end
+
+  end
+
+  describe '#abs_get_resource_hosts' do
+
+    context 'when a valid response is returned' do
+
+      it 'sets the environment variable, logs the hosts, and returns the response(s)' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(test_http_response)
+
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_AWSDIRECT_RESPONSE_BODY)
```

Comment:
All three of these don't actually happen in the method under test.  So if the stuff mentioned above gets mocked, these all go away.